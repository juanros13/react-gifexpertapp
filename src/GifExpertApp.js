import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {
    //const categories = ['One punch','Samurai X','Dragon ball']


    const [categories, setCategories] = useState(['Dragon ball']);
    
    // const hadledAdd = () =>{
    //     setCategories([...categories, 'HunterXHunter']);
    //     setCategories( cats => [...cats, 'HunterXHunter']);
    // }

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories = { setCategories }/>
            <hr/>
            {/* <button onClick={ hadledAdd }>Agregar</button> */}
            <ol>
                {
                    categories.map(category =>
                        <GifGrid 
                            category = { category }
                            key = { category }
                        
                        />
                    )
                }
            </ol>
           
        </>
    );
}
