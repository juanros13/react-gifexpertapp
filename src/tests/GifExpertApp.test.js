import React from 'react';
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";
import '@testing-library/jest-dom';


describe('Puebas con el componente <GifExpertApp />', () => {
    test('Debe mostrar correctamente el GifExpertApp', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    });
    test('debe de mostrar una lista de categorias', () => {
        const categories = ['one punch', 'attack on titan'];
        const wrapper = shallow(<GifExpertApp defaultCategories = { categories }/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);


    });
    
    
    
})
