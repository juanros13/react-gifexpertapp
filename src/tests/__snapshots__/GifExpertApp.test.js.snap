// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas con el componente <GifExpertApp /> Debe mostrar correctamente el GifExpertApp 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Puebas con el componente <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="one punch"
      key="one punch"
    />
    <GifGrid
      category="attack on titan"
      key="attack on titan"
    />
  </ol>
</Fragment>
`;
